starknet/test/eth-sig-types.ts

```diff
@@ -0,0 +1,94 @@
+export const sharedTypes = {
+  Strategy: [
+    { name: 'address', type: 'uint256' },
+    { name: 'params', type: 'uint256[]' },
+  ],
+  IndexedStrategy: [
+    { name: 'index', type: 'uint256' },
+    { name: 'params', type: 'uint256[]' },
+  ],
+};
+
+export const proposeTypes = {
+  Propose: [
+    { name: 'authenticator', type: 'uint256' },
+    { name: 'space', type: 'uint256' },
+    { name: 'author', type: 'address' },
+    { name: 'executionStrategy', type: 'Strategy' },
+    { name: 'userProposalValidationParams', type: 'uint256[]' },
+    { name: 'metadataUri', type: 'uint256[]' },
```

Comment:
since the `space contract parity` PR got merged, `metadataUri` should be right after `author` (only for `propose`, not for `vote` or `updateProposal`